software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakClusterImpl.java

```diff
@@ -79,7 +82,7 @@ public void start(Collection<? extends Location> locations) {
 
         //FIXME: add a quorum to tolerate failed nodes before setting on fire.
         Optional<Entity> anyNode = Iterables.tryFind(getMembers(), Predicates.and(
-                Predicates.instanceOf(RiakNode.class),
+                Predicates.or(Predicates.instanceOf(RiakNode.class), Predicates.instanceOf(SameServerEntity.class)),
                 EntityPredicates.attributeEqualTo(RiakNode.RIAK_NODE_HAS_JOINED_CLUSTER, true),
```

Comment:
The entity assumes that the members are RiakNodes, thus relaying on specific behaviour from them. For example `RIAK_NODE_HAS_JOINED_CLUSTER` will be set by the node, it's not managed by the cluster. Another example is calling the effector `REMOVE_FROM_CLUSTER` on the members.